### PR TITLE
Refactor Apps chatflows

### PR DESCRIPTION
**Pull request by dmahmouali**
issue: https://github.com/threefoldtech/js-sdk/issues/502

### CODE REVIEW

File: jumpscale/sals/marketplace/chatflow.py [hunk 1]
```diff
@@ -29,3 +37,147 @@ def _validate_user(self):
...
+                currency=self.currency,
+                **self.query,
+            )
+
+        if not self.pool_info:
+            raise StopChatFlow(f"Failed to deploy solution {self.pool_info}")
+        self.pool_id = self.pool_info.reservation_id
+        return self.pool_id
+
+    def _deploy_network(self):
```
Comment (m-motawea) [inline jumpscale/sals/marketplace/chatflow.py:93]: most of these methods are required for apps chatflows only not all marketplace solutions. can we apps inherit from a different class which includes these stuff?

```python
class AppChatflow(MarketplaceChatflow):
```

Comment (dmahmouali) [reply jumpscale/sals/marketplace/chatflow.py]: Added `MarketPlaceAppsChatflow` base class and use it in apps.
Thanks @m-motawea

**Comment by waleedhammam** [comment]
make sure to have https://github.com/threefoldtech/js-sdk/pull/508 merged in here before we merge into `development_capacitypool_apps`

File: jumpscale/sals/marketplace/apps_chatflow.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+from jumpscale.core.base import StoredFactory
+from jumpscale.loader import j
+from .solutions import solutions
+from .deployer import deployer
+from .chatflow import MarketPlaceChatflow
+from jumpscale.packages.marketplace.bottle.models import UserEntry
+from jumpscale.sals.chatflows.chatflows import StopChatFlow, chatflow_step
+
+
+FARM_NAMES = ["freefarm"]
```
Comment (waleedhammam) [inline jumpscale/sals/marketplace/apps_chatflow.py:13]: I think it's time to pick from all farms, or still they have issues ?

File: jumpscale/packages/marketplace/chats/peertube.py [hunk 3]
```diff
@@ -293,13 +112,35 @@ def reservation(self):
...
+            pool_id=self.pool_info.reservation_id,
+            gateway_id=self.gateway.node_id,
+            network_name=self.network_view.name,
+            trc_secret=self.secret,
+            domain=self.domain,
+            email=self.user_info()["email"],
+            solution_ip=self.ip_address,
+            solution_port=80,
+            enforce_https=True,
+            test_cert=True,
```
Comment (waleedhammam) [inline jumpscale/packages/marketplace/chats/peertube.py:125]: I think we can make the default `test_cert=False`
= True only for debugging purposes

**Pull request merged**